lightning/src/routing/gossip.rs

```diff
@@ -1046,23 +1046,50 @@ pub struct NodeAnnouncementInfo {
 	/// May be invalid or malicious (eg control chars),
 	/// should not be exposed to the user.
 	pub alias: NodeAlias,
-	/// Internet-level addresses via which one can connect to the node
-	pub addresses: Vec<NetAddress>,
 	/// An initial announcement of the node
 	/// Mostly redundant with the data we store in fields explicitly.
 	/// Everything else is useful only for sending out for initial routing sync.
 	/// Not stored if contains excess data to prevent DoS.
 	pub announcement_message: Option<NodeAnnouncement>
 }
 
-impl_writeable_tlv_based!(NodeAnnouncementInfo, {
-	(0, features, required),
-	(2, last_update, required),
-	(4, rgb, required),
-	(6, alias, required),
-	(8, announcement_message, option),
-	(10, addresses, vec_type),
-});
+impl NodeAnnouncementInfo {
+	/// Internet-level addresses via which one can connect to the node
+	pub fn addresses(&self) -> Vec<NetAddress> {
+		self.announcement_message.as_ref()
+			.map(|msg| msg.contents.addresses.clone())
```

Comment:
I think there is no way to return `&Vec<NetAddress>` because the default empty vec would be a temporary local reference. Alternatives that work are `Option<&Vec<NetAddress>>` or `&[NetAddress]`.

I would lean into `Option<&Vec<NetAddress>>`